### ISSUE TITLE
implement alert severity (warning, critical, exception, notification)

### ISSUE THREAD

**Issue by kklemen** [target/goalert#46]
**Is your feature request related to a problem? Please describe:**
There is currently no option to specify alert severity which is major drawback because alerts are all considered to be of equal importance. 
In production environments alerts are delegated by priority; critical alerts like server malfunction need to be taken care of immediately while low priority alerts like a certificate expiring within a month or log folder growing to large size do not require immediate attention and can be handled during maintenance hours days or even weeks later.

**Describe the solution you'd like:**
To be able to specify alert severity via parameter (when sending an alert) and then route an alert to an appropriate channel(s).
example:
low severity for alerts that can be handled within hours/days: send over email because there is no point waking up oncall staff at 2AM in the morning over trivial issues.
high severity for alerts that must be handled immediately: send over sms (and in addition email, slack etc)

**Describe alternatives you've considered:**
There is no alternative as severity is not implemented.

**Comment by donna-roisen** [comment]
Thank you for your interest in GoAlert.  We are glad to hear from you and consider the inputs we receive from users of GoAlert as an opportunity to improve the product.

GoAlert has been designed to serve the use case for teams and/or on-call person(s) to get alerted for critical / high severity alerts.

We have thought quite a bit about use cases for non-critical alerts and have intentionally not approached these use cases.  We believe deeply in the Unix philosophy as summarized by Peter Salus '...do one thing and do it well."  For GoAlert, this means that we are laser focused on the use case for processing critical alerts that require immediate attention by a person.  Down the road, it's possible we'll come up with a solution for non-critical alerts...but there's quite a bit of work left to do within our current use case.

With all that said, there are a few ways that you could use GoAlert today to handle non-critical alerts.   One option is you could create a separate service and escalation policy for processing non-critical alerts.  On this escalation policy, you could use the Slack integration to send notifications of these non-critical alerts to a channel of your choice.  Slack notifications are likely less intrusive that SMS/Voice notifications.

We'll keep this issue open for comments from the broader community on if/how non-critical alert processing use cases should be incorporated into GoAlert.

**Comment by kklemen** [comment]
There's so much wrong with your reply. Seems like you do not understand production systems at all.

An efficient alerting system needs to accept alerts from multiple sources (check)

- aggregate alerts in one place (goalert has beautiful gui in one panel (gui, check)
- must deduplicate alerts to prevent alert-fatigue 
-have a way to prioritize alerts so critical alerts are sorted first (huge fail, you're saying you INTENTIONALLY only support critical alerts when you already built the rest of the foundation to support it? why this decision? it makes no sense, you're intentionally crippling an awesome system)
-have a way to delegate alerts between support staff (so multiple people dont work on the same issue, check)

So what you're suggesting is to NOT use a single alerting system (goalert) but instead use two systems where I first have to use one system to split alerts by severity and then route only critical alerts to goalert. 
It adds multiple points of failure for more thing to go wrong.
it adds more points of configuration so it becomes more complicated and harder to maintain.

This is not efficient at all. the very opposite. You're saying goalert is a luxury racing car without a steering wheel?

Already now go-alert isnt doing one thing, its doing multiple so i dont know what you mean by laser targeting (are you trying to shoot yourself in the foot by building a highly niche product? is this the opinion of the entire team?)

You got gui aggregation of alerts, you got multiple alerting outputs, you even got support for alerts being created by incoming email which is an extremely niche feature compared to alert severity and adding the first but not the latter makes no sense to me.

In other words, if you want to be the next victorops or pagerduty you MUST add alert severity or goalert will not move beyond a hobby project.  I'd pay/donate 100-200 euro right now if alert severity and email routing was implemented because everything else (the hard part) is already done and its done beautifully. But as of right now goalert is utterly useless to me. Maybe for managing one or two vpss sure, but for my use case managing 50+ servers with 10+ senior support staff it's utterly useless until the very point where these two features are added. then its immediately the next best thing since the invention of sliced bread.

**Comment by pnengchu** [comment]
closing due to inactivity there are no plans to implement this currently.
